implementingSwashbuckle.md

```diff
@@ -0,0 +1,133 @@
+
+```
+//Enables Swashbuckle and related Swagger options.
+ private void generateSwagger(HttpConfiguration config)
+ {
+     config.EnsureInitialized();
+     var swaggerProvider = new SwaggerGenerator(
+     config.Services.GetApiExplorer(),
+     config.Formatters.JsonFormatter.SerializerSettings,
+     new Dictionary<string, Info> { { "v1", new Info { version = "v1", title = "My API",
+     description = "Provides an interface between our software and other software. This API
+     sends requests to verify things."} }},
+     new SwaggerGeneratorOptions(
+          // Apply your Swagger options here.
+             schemaIdSelector: (type) => type.FriendlyId(true),
+          //Implements the SwaggerTitleFilter class, which generates "title" members in the
+          //"definitions" model of the swagger.json output.
+             modelFilters: new List<IModelFilter>(){ new SwaggerTitleFilter() },
+             conflictingActionsResolver: (apiDescriptions) => apiDescriptions.GetEnumerator().Current,
+          //Enables schema filters, such as recognizing XML comments, including response codes,
+          //and adding operation filters.
+             schemaFilters: new List<ISchemaFilter>() {
+                 new ApplySwaggerSchemaFilterAttributes()
+                 },
+             operationFilters: new List<IOperationFilter>() {
+              //Recognize XML comments in the code and write them to the MyAPI.XML file.
+                 new ApplyXmlActionComments("MyAPI.XML"),
+              //Enable response code classes and attributes through Swashbuckle.
+                 new ApplySwaggerResponseAttributes(),
+              //Modify operation descriptions after they are generated.
+                 new ApplySwaggerOperationAttributes(),
+              //Enable operation filters and attributes.
+                 new ApplySwaggerOperationFilterAttributes()
+             }
+
+         )
+
+     );
+     var swaggerDoc = swaggerProvider.GetSwagger("http://localhost:8006", "v1");
+
+  // Serializes the swaggerDoc variable.
+     var swaggerString = JsonConvert.SerializeObject(
+         swaggerDoc,
+         Formatting.Indented,
+         new JsonSerializerSettings
+         {
+             NullValueHandling = NullValueHandling.Ignore,
+             Converters = new[] { new VendorExtensionsConverter() }
+         }
+     );
+
+```
+## Using DapperDox
+
+[DapperDox](http://dapperdox.io/) is an open source documentation framework for OpenAPI specifications. Instead of having Swashbuckle publish our API specificaiton in the Swagger UI, I added the following code to the ```Startup.cs``` file. This code writes the Swagger specification to a swagger.json file.
+
```

Comment:
Not required, but you can use markdown markup (ha) with `swagger.json` for all file names, helps pick it out in the text.